homeassistant/components/esphome/voice_assistant.py

```diff
@@ -302,11 +304,32 @@ async def _send_tts(self, media_id: str) -> None:
                 VoiceAssistantEventType.VOICE_ASSISTANT_TTS_STREAM_START, {}
             )
 
-            _extension, audio_bytes = await tts.async_get_media_source_audio(
+            extension, data = await tts.async_get_media_source_audio(
                 self.hass,
                 media_id,
             )
 
+            if extension != "wav":
+                raise ValueError(f"Only WAV audio can be streamed, got {extension}")
+
+            with io.BytesIO(data) as wav_io:
+                with wave.open(wav_io, "rb") as wav_file:
```

Comment:
You can combine with-statements. 
```python
            with io.BytesIO(data) as wav_io, wave.open(wav_io, "rb") as wav_file:
```

Reply:
Not in this case without extra care. The `wave.open` context manager needs to close before it flushes data to the `BytesIO`.